crowdsourcing/servermgr/app.py

```diff
@@ -69,6 +70,18 @@
     run_sh_gz_b64 = b64encode(gzip.compress(txt)).decode("utf-8")
     run_flat_sh_gz_b64 = b64encode(gzip.compress(txt_flat)).decode("utf-8")
 
+def register_dashboard_subdomain(cf, zone_id, ip):
+    """Registers a unique subdomain for craftassist.io
+    that serves proxied HTTP content using cloudflare.
+
+    Args:
+    cf -- CloudFlare context with R/W permissions.
+    zone_id -- zone ID used to locate DNS records.
+    ip -- IP of the ECS container that runs dashboard.
+    """
+    subdomain_name = "dashboard-{}".format(randint(0, 10 ** 9))
```

Comment:
randint can technically produce the same number twice in rare cases. So before doing `dns_records.post`, you have to check that the record doesn't exist (if it exists, choose a different random number)